### DISCUSSION TITLE
Get list of faces that an isoline passes through?

### DISCUSSION THREAD

**Discussion by rahul-mitra13** [libigl/libigl#2359] [Q&A] [answered]
In my particular application, I need to know the list of faces that a given isoline passes through. So far, the results from igl::isolines has been great in generating the isoline as a curve on my surface! I was wondering if there's anyway to also query the list of faces that an isoline, given a particular isovalue, passes through? Would appreciate any pointers on where to look/what to modify in isolines.cpp? Any feedback provided would be helpful. :)

**Comment by alecjacobson** [comment, marked as answer]
I guess the easy way to do this is to just loop

```cpp
std::vector<int> passes;
for(int f = 0;f<F.rows();f++)
{
  if(( 
        S(F(f,0))<iso ||
        S(F(f,1))<iso ||
        S(F(f,2))<iso)
      &&
      (
        S(F(f,0))>iso ||
        S(F(f,1))>iso ||
        S(F(f,2))>iso))
  {
    passes.push_back(f);
  }
}
```

**Reply by rahul-mitra13** [reply]
Oh yea of course! 😅 Thank you
